usaspending_api/data_load/fpds_loader.py

```diff
@@ -0,0 +1,314 @@
+            recipient_location[key] = recipient_location_functions[key](broker_object)
+
+        connected_objects["recipient_location"] = recipient_location
+
+        # legal entity
+        legal_entity = {}
+        for key in legal_entity_columns:
+            legal_entity[key] = capitalize_if_string(broker_object[legal_entity_columns[key]])
+
+        for key in legal_entity_boolean_columns:
+            legal_entity[key] = false_if_null(broker_object[legal_entity_boolean_columns[key]])
+
+        for key in legal_entity_functions:
+            legal_entity[key] = legal_entity_functions[key](broker_object)
+
+        connected_objects["legal_entity"] = legal_entity
+
+        # place_of_performance_location
+        place_of_performance_location = {}
+        for key in place_of_performance_columns:
+            place_of_performance_location[key] = capitalize_if_string(broker_object[place_of_performance_columns[key]])
+
+        for key in place_of_performance_functions:
+            place_of_performance_location[key] = place_of_performance_functions[key](broker_object)
+
+        connected_objects["place_of_performance_location"] = place_of_performance_location
+
+        # matching award. NOT a real db object, but needs to be stored when making the link in load_transactions
+        connected_objects["generated_unique_award_id"] = broker_object["unique_award_key"]
+
+        # award. NOT used if a matching award is found later
+        award = {}
+        for key in award_functions:
+            award[key] = award_functions[key](broker_object)
+        connected_objects["award"] = award
+
+        # transaction_normalized
+        transaction_normalized = {}
+        for key in transaction_normalized_columns:
+            transaction_normalized[transaction_normalized_columns[key]] = capitalize_if_string(broker_object[key])
+
+        for key in transaction_normalized_functions:
+            transaction_normalized[key] = transaction_normalized_functions[key](broker_object)
+
+        connected_objects["transaction_normalized"] = transaction_normalized
+
+        # transaction_fpds
+        transaction_fpds = {}
+        for key in transaction_fpds_columns:
+            transaction_fpds[transaction_fpds_columns[key]] = capitalize_if_string(broker_object[key])
+
+        for key in transaction_fpds_boolean_columns:
+            transaction_fpds[transaction_fpds_boolean_columns[key]] = false_if_null(broker_object[key])
+
+        for key in transaction_fpds_functions:
+            transaction_fpds[key] = transaction_fpds_functions[key](broker_object)
+
+        connected_objects["transaction_fpds"] = transaction_fpds
+
+        retval.append(connected_objects)
+    return retval
+
+
+def load_transactions(load_objects):
+    with psycopg2.connect(dsn=USASPENDING_CONNECTION_STRING) as connection:
+        with connection.cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
+            for load_object in load_objects:
```

Comment:
This is still loading data row-by-row?!?!? My understanding was to move into loading "blocks" of transactions for performance. It appears the main difference between this script and the current is the Django ORM is no longer leveraged. I don't expect this to help performance much

Reply:
Empirically, it works faster; I don't fully understand why. I wanted to have cursors execute in batches, but the need to do things like line 195 (which I have to save the location reference for the legal entity, then see what primary key was generated to add it to the legal_entity) makes that more difficult. It would be possible to lay down all the objects then do linking afterwards, but that might require relaxing foreign key constraints.